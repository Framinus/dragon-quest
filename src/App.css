.App {
  text-align: center;
}

.main-title {
  margin-bottom: 40px;
}

.dragon-collection-description {
  margin: 20px;
}

.dragon-card-container {
  background-color: #000;
  border: 1px solid grey;
  border-radius: 10px;
  color: #fff;
  font-size: 18px;
  height: 400px;
  margin: 20px auto;
  padding: 20px 10px;
  width: 300px;
}

.dragon-card-image {
  border: 1px solid black;
  border-radius: 5px;
  height: 150px;
  width: 250px;
}

.control-btns .btn {
  margin: 0 4px;
}
