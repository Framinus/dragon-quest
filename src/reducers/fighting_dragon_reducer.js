import { FIGHTING_DRAGON } from '../actions';

export default function(state={}, action) {
  switch(action.type) {
    case FIGHTING_DRAGON:
      return action.payload
    default: return state
  }
}
