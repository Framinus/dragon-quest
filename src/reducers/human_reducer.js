import { SPAWN_HUMAN, SAVE_NEW_HUMAN } from '../actions';

export default function (state={}, action) {
  switch(action.type) {
    case SPAWN_HUMAN:
      return action.payload.data.human
    case SAVE_NEW_HUMAN:
      return action.payload
    default: return state;
  }
}
