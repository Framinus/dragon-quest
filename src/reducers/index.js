import { combineReducers } from 'redux';
import DragonsReducer from './reducer_dragons';
import RandomDragonReducer from './reducer_random_dragon';
import EnterFightModeReducer from './reducer_enter_fight';
import FightingDragonReducer from './reducer_fighting_dragon';
import HumanReducer from './reducer_human';
import MergingDragonsReducer from './reducer_merging_dragons';

const rootReducer = combineReducers({
  dragons: DragonsReducer,
  fightMode: EnterFightModeReducer,
  fightingDragon: FightingDragonReducer,
  human: HumanReducer,
  mergingDragons: MergingDragonsReducer,
  randomDragon: RandomDragonReducer,
});

export default rootReducer;
