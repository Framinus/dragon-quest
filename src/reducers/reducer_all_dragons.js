import { GET_DRAGON_LIST } from '../actions';

export default function reducerAllDragons(state=[], action) {
  switch(action.type) {
    case GET_DRAGON_LIST:
      return action.payload
    default: return state;
  }
}
