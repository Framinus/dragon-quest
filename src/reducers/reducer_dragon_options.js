import { DRAGON_BY_ID } from '../actions';

export default function(state={}, action) {
  switch(action.type) {
    case DRAGON_BY_ID:
      return action.payload.data;
    default: return state
  }
}
