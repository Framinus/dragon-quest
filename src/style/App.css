.App {
  text-align: center;
}

.main-title {
  margin-bottom: 20px;
}

.dragon-collection-description {
  margin: 10px;
}

.dragon-card-container {
  background-color: #000;
  border: 1px solid grey;
  border-radius: 5px;
  color: #fff;
  font-size: 18px;
  height: 400px;
  margin: 10px auto;
  padding: 20px 10px;
  width: 300px;
}

.dragon-card-image {
  border: 1px solid black;
  border-radius: 5px;
  height: 150px;
  width: 250px;
}

.control-btns .btn {
  margin: 0 4px;
}



@media (min-width: 600px) {

  .dragon-collection-description {
    margin: 20px;
  }

  .dragon-card-container {
    margin: 20px auto;
  }

  .main-title {
    margin-bottom: 40px;
  }

}
