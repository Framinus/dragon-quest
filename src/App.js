import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css';
import CardCarousel from './containers/card_carousel';
import GetRandomDragon from './containers/get_random_dragon';
import Fight from './containers/fightMode';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      randomDragonVisible: false,
      dragonCollectionVisible: true,
      fightMode: false,
    }

    this.callDragon = this.callDragon.bind(this);
    this.acceptDragon = this.acceptDragon.bind(this);
    this.toggleFightMode = this.toggleFightMode.bind(this);

  }

  callDragon() {
    this.setState({
      randomDragonVisible: true,
    })
  }

  acceptDragon() {
    this.setState({
      randomDragonVisible: false,
    })
  }

  toggleFightMode() {
    if (this.state.fightMode) {
      this.setState({
        fightMode: false,
      })
    } else {
      this.setState({
        fightMode: true,
      })
    }
  }

  render() {
    let mainView = null;
    let randomDragon = null;

    if (this.state.randomDragonVisible) {
      randomDragon = (
        <div className="dragon-selection-container">
          <GetRandomDragon
            acceptDragon={this.acceptDragon}
          />
            <div className="dragon-selection-description">
          </div>
        </div>
      )
    }

    if (this.state.fightMode) {
      mainView = (
        <div className="fight-screen-container">
          <Fight
            toggleFightMode={this.toggleFightMode}
            fightMode={this.state.fightMode}
          />
        </div>
      )
    } else {
      mainView = (
        <div>
          <h1>Dragon Quest</h1>
          <div className="dragon-collection-description">
            <p>Call a dragon and accept her to add her to your defense squad, or send her away to try for another type! You can only have ten at any given time. Click "fight" on a dragon's card to face off against a human that blunders into the village. Each successive human you face might be a little bit harder. You will level up after a successful battle, or you can merge two dragons for a chance to create a powerful new dragon.</p>
            <button id="dragon-call-btn" className="btn btn-success" onClick={this.callDragon}>
              Call Dragon
            </button>
            <button id="dragon-merge-btn" className="btn btn-info">
              Merge Dragons
            </button>
          </div>
          <div className="carousel">
            <CardCarousel
              toggleFightMode={this.toggleFightMode}
            />
          </div>
          {randomDragon}
        </div>
      )
    }

    return (
      <div className="App">
        {mainView}
      </div>
    );
  }
}

const mapStateToProps = ({ fightMode }) => {
  return { fightMode };
};

export default connect(mapStateToProps)(App);
