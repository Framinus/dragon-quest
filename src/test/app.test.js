import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import enzyme, { shallow, mount, render } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import promise from 'redux-promise';
import renderer from 'react-test-renderer';
import toJSON from 'enzyme-to-json';
import ConnectedApp from '../App';
import reducers from '../reducers';
import { App } from '../App';

enzyme.configure({ adapter: new Adapter() });

const createStoreWithMiddleware = createStore(
  reducers,
  compose(
    applyMiddleware(promise))
  );

describe('App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={createStoreWithMiddleware} >
        <ConnectedApp />
      </Provider>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  const wrapper = shallow(< App />);

  it('should be selectable by class "App"', () => {
    expect(wrapper.is('.App')).toBe(true);
  });

  it('should render a MainView component', () => {
    expect(wrapper.find('.MainView')).toHaveLength(1);
  });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <Provider store={createStoreWithMiddleware} >
        <ConnectedApp />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})
