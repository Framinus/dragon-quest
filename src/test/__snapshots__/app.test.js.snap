// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match its empty snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="main-title"
  >
    Dragon Quest
  </h1>
  <div
    className="main-view-container"
  >
    <div
      className="dragon-collection-description"
    >
      <div
        className="control-btns"
      >
        <span>
          <button
            className="btn btn-success"
            id="dragon-call-btn"
            onClick={[Function]}
          >
            Call Dragon
          </button>
        </span>
        <span>
          <button
            className="btn btn-info"
            id="dragon-merge-btn"
            onClick={[Function]}
          >
            Merge Mode
          </button>
        </span>
      </div>
    </div>
    <div
      className="carousel"
    >
      <div>
        <div>
          <h4>
            No dragons here!
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
